pkg/crdexplorer/customizedexplorer/customized.go

```diff
@@ -49,14 +57,228 @@ func (e *CustomizedExplorer) HookEnabled(kind schema.GroupVersionKind, operation
+// return matched value to indicate whether there is a matching hook.
+func (e *CustomizedExplorer) GetDependencies(ctx context.Context, attributes *webhook.RequestAttributes) (dependencies []configv1alpha1.DependentObjectReference, matched bool, err error) {
+	if !e.hookManager.HasSynced() {
+		err = fmt.Errorf("not yet ready to handle request")
+		return
+	}
+
+	var response *webhook.ResponseAttributes
+	response, matched, err = e.explore(ctx, attributes)
+	if err != nil {
+		return
+	}
+	if !matched {
+		return
+	}
+
+	return response.Dependencies, matched, nil
+}
+
+// PackObject return the patch body that package resource template with request object.
+// return matched value to indicate whether there is a matching hook.
+func (e *CustomizedExplorer) PackObject(ctx context.Context, attributes *webhook.RequestAttributes) (patch []byte, patchType configv1alpha1.PatchType, matched bool, err error) {
+	if !e.hookManager.HasSynced() {
+		err = fmt.Errorf("not yet ready to handle request")
+		return
+	}
+
+	var response *webhook.ResponseAttributes
+	response, matched, err = e.explore(ctx, attributes)
+	if err != nil {
+		return
+	}
+	if !matched {
+		return
+	}
+
+	return response.Patch, response.PatchType, matched, nil
 }
 
 // GetHealthy tells if the object in healthy state.
 // return matched value to indicate whether there is a matching hook.
-func (e *CustomizedExplorer) GetHealthy(ctx context.Context, operation configv1alpha1.OperationType,
-	object runtime.Object) (healthy, matched bool, err error) {
-	return false, false, err
+func (e *CustomizedExplorer) GetHealthy(ctx context.Context, attributes *webhook.RequestAttributes) (healthy, matched bool, err error) {
+	if !e.hookManager.HasSynced() {
+		err = fmt.Errorf("not yet ready to handle request")
+		return
+	}
+
+	var response *webhook.ResponseAttributes
+	response, matched, err = e.explore(ctx, attributes)
+	if err != nil {
+		return
+	}
+	if !matched {
+		return
+	}
+
+	return response.Healthy, matched, nil
+}
+
+func (e *CustomizedExplorer) getFirstRelevantHook(attributes *webhook.RequestAttributes) configmanager.WebhookAccessor {
+	relevantHooks := make([]configmanager.WebhookAccessor, 0)
+	for _, hook := range e.hookManager.HookAccessors() {
+		matcher := &crdexplorerutil.Matcher{
+			Operation: configv1alpha1.ExploreReplica,
+			Object:    attributes.Object,
+			Resource:  attributes.Resource,
+		}
+		if shouldCallHook(hook, matcher) {
+			relevantHooks = append(relevantHooks, hook)
+		}
+	}
+
+	if len(relevantHooks) == 0 {
+		return nil
+	}
+
+	// Sort relevantHooks alphabetically, taking the first element for spawning remote calls
+	sort.SliceStable(relevantHooks, func(i, j int) bool {
+		return relevantHooks[i].GetUID() < relevantHooks[j].GetUID()
+	})
+	return relevantHooks[0]
+}
+
+func (e *CustomizedExplorer) explore(ctx context.Context, attributes *webhook.RequestAttributes) (*webhook.ResponseAttributes, bool, error) {
+	var matched bool
```

Comment:
no need this variable, will generally be true except for getFirstRelevantHook return value is nil.